build/common/installer/scripts/tomlparser.rb

```diff
@@ -227,6 +230,29 @@ def populateSettingValuesFromConfigMap(parsedConfig)
     rescue => errorStr
       ConfigParseErrorLogger.logError("Exception while reading config map settings for adx database name - #{errorStr}, using default #{@adxDatabaseName}, please check config map for errors")
     end
+
+    #Get Kubernetes Metadata setting
+    begin
+      if !parsedConfig[:log_collection_settings][:metadata_collection].nil? && !parsedConfig[:log_collection_settings][:metadata_collection][:enabled].nil?
+        puts "config::Using config map setting for kubernetes metadata"
+        @logEnableKubernetesMetadata = parsedConfig[:log_collection_settings][:metadata_collection][:enabled]
+        if !parsedConfig[:log_collection_settings][:metadata_collection][:include_fields].nil?
+          puts "config::Using config map setting for kubernetes metadata include fields"
+          include_fields = parsedConfig[:log_collection_settings][:metadata_collection][:include_fields]
```

Comment:
1.  we should validate the provided field in the include_fields is supported list or not? If its not in supported list, log warn message

Reply:
if the include_fields does not match the data will be dropped

Reply:
warning added